### PR TITLE
Update dqlite and deps

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
## Description
This PR updates dqlite and it's dependencies, go-dqlite and sqlite.

### Dqlite updates:
Dqlite [1.16.7](https://github.com/canonical/dqlite/releases/tag/v1.16.7) has just been released (and contains a fix for a segfault which affects other PRs).
The latest dqlite is using multiple threads to serve queries.

### Dqlite dependecies
For the dqlite dependencies we choose to be in sync with the libraries of the Ubuntu Noble LTS where possible.
Example: musl `v1.2.4` ([launchpad](https://launchpad.net/ubuntu/noble/amd64/musl/1.2.4-2))

### K8s-dqlite is updated
The same updates have already been merged in k8s-dqlite: https://github.com/canonical/k8s-dqlite/pull/151 and https://github.com/canonical/k8s-dqlite/pull/153

### Arm64
The Juju team discovered segfault issues building dqlite on arm64 machines, this flag addresses the issues: `-Wl,-z,stack-size=1048576`
See: https://wiki.musl-libc.org/functional-differences-from-glibc.html#Thread-stack-size

### CODE REVIEW

File: src/k8s/hack/env.sh [hunk 1]
```diff
@@ -11,11 +11,12 @@ REPO_SQLITE="https://github.com/sqlite/sqlite.git"
...
+TAG_MUSL="v1.2.4"
 TAG_LIBTIRPC="upstream/1.3.3"
-TAG_LIBNSL="v2.0.0"
+TAG_LIBNSL="v2.0.1"
 TAG_LIBUV="v1.48.0"
 TAG_LIBLZ4="v1.9.4"
-TAG_RAFT="v0.18.0"
-TAG_SQLITE="version-3.40.0"
-TAG_DQLITE="v1.15.1"
+TAG_RAFT="v0.18.1"
```
Comment (neoaggelos) [inline src/k8s/hack/env.sh:19]: are we dropping raft? if so, lets remove this entirely and make it all into a single PR

**Review by neoaggelos — COMMENTED** [review]
Note that we need the same PR on the k8s-dqlite repo

**Comment by louiseschmidtgen** [comment]
> Note that we need the same PR on the k8s-dqlite repo

I've already merged the update on k8s-dqlite side: https://github.com/canonical/k8s-dqlite/pull/151

File: src/k8s/hack/env.sh (hunk 1 above)
Comment (louiseschmidtgen) [reply src/k8s/hack/env.sh:19]: I thought I'd put up a separate PR to drop raft, but I can add it here.

**Review by neoaggelos — COMMENTED** [review]
If `musl` is in ubuntu 24.04 repos, avoid building it from source. this might have to wait until we move to `core24` base

File: build-scripts/components/k8s-dqlite/version [hunk 2]
```diff
@@ -1 +1 @@
-master
+marco6/improve-compaction
```
Comment (neoaggelos) [inline build-scripts/components/k8s-dqlite/version:1]: that should be reverted

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**